### PR TITLE
fix(cli): guard oversized contract publish

### PR DESCRIPTION
**Pull request by unclekaldoteth**
### Description
Add contract-publish edge-case coverage:
- CLI now rejects oversized publish transactions with a clear error.
- New tests for invalid contract name, non-ASCII contract source, and oversized contract source.
- Add a clarity analysis test that rejects a Clarity4-only builtin when analyzed under Clarity2 (version mismatch).

### Applicable issues
- fixes #

### Additional info (benefits, drawbacks, caveats)
CLI-only guard; no consensus behavior changes. The size check fails fast before signing if the unsigned transaction exceeds `MAX_TRANSACTION_LEN`.

Test evidence:
$ cargo test -p stacks-cli
running 21 tests
test test::simple_addresses ... ok
test test::generate_should_work ... ok
test test::simple_decode_header ... ok
test test::simple_decode_block ... ok
test test::custom_chain_id ... ok
test test::simple_token_transfer ... ok
test test::test_contract_call_with_serialized_arg_from_file_fails_due_to_file ... ok
test test::test_contract_call_with_serialized_arg_from_file_fails_due_to_short_buffer ... ok
test test::test_contract_call_with_serialized_arg_from_file_fails_due_to_bad_hex ... ok
test test::test_contract_publish_fails_with_non_ascii_contract_source ... ok
test test::test_contract_publish_fails_with_invalid_name ... ok
test test::simple_decode_tx ... ok
test test::test_contract_call_with_serialized_arg_from_file_ok ... ok
test test::simple_cc ... ok
test test::test_contract_publish_fails_with_oversized_source ... ok
test test::test_contract_publish_fails_on_unexistent_file ... ok
test test::test_contract_publish_fails_with_postcond_mode ... ok
test test::test_contract_publish_fails_with_anchor_mode ... ok
test test::test_contract_publish_ok_with_anchor_mode ... ok
test test::test_contract_publish_ok_with_mandatory_params ... ok
test test::test_contract_publish_ok_with_postcond_mode ... ok

test result: ok. 21 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.03s

$ cargo test -p stackslib test_contract_hash_rejected_in_clarity2
running 1 test
test clarity_vm::tests::contracts::test_contract_hash_rejected_in_clarity2 ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 1819 filtered out; finished in 0.04s

### Checklist
- [x] Test coverage for new or modified code paths
- [x] For new Clarity features or consensus changes, add property tests (N/A)
- [ ] Changelog is updated (tests/CLI guard only)
- [x] Required documentation changes (N/A)
- [x] New clarity functions have corresponding PR in `clarity-benchmarking` repo (N/A)

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -17,6 +17,7 @@
 ### Checklist
 
 - [ ] Test coverage for new or modified code paths
+- [ ] For new Clarity features or consensus changes, add property tests (see [`docs/property-testing.md`](/docs/property-testing.md))
 - [ ] Changelog is updated
-- [ ] Required documentation changes (e.g., `docs/rpc/openapi.yaml` and `rpc-endpoints.md` for v2 endpoints, `event-dispatcher.md` for new events)
+- [ ] Required documentation changes (e.g., [`rpc/openapi.yaml`](/docs/rpc/openapi.yaml) for RPC endpoints, [`event-dispatcher.md`](/docs/event-dispatcher.md) for new events)
```
Comment (wileyj) [inline .github/PULL_REQUEST_TEMPLATE.md:22]: https://github.com/stacks-network/stacks-core/pull/6873#discussion_r2794202042

**Comment by jcnelson** [comment]
Closing due to inactivity
